### PR TITLE
issues: Obsolete branch specification when creating #83

### PR DESCRIPTION
**Pull request by yoshio15**
## 💪 Summary
fix the issue #83 .
- add getting default branch function
- and set it to `s:gh_issue_new.branch`

## 🏁 Goals
- to be able to create new issue by following command. (removed branch name)
```
:e gh://:owner/:repo/issues/new
```

### CODE REVIEW

File: autoload/gh/github/repos.vim [hunk 1]
```diff
@@ -51,3 +51,12 @@ function! gh#github#repos#create(data) abort
...
   return gh#http#request(settings)
 endfunction
+
+function! gh#github#repos#get_repo(owner, repo) abort
+  let url = printf('https://api.github.com/repos/%s/%s', a:owner, a:repo)
+  let settings = {
+        \ 'method': 'GET',
+        \ 'url': url,
+        \ }
+  return gh#http#request(settings)
```
Comment (skanehira) [inline autoload/gh/github/repos.vim]: If just request GET, you can use `gh#http#get()` to get it.

File: autoload/gh/issues.vim [hunk 2]
```diff
@@ -194,17 +194,31 @@ function! gh#issues#new() abort
   call gh#gh#init_buffer()
   call gh#gh#set_message_buf('loading')
 
-  let m = matchlist(bufname(), 'gh://\(.\{-}\)/\(.\{-}\)/\(.*\)/issues/new$')
+  let m = matchlist(bufname(), 'gh://\(.\{-}\)/\(.\{-}\)/issues/new$')
+
+  let l:owner = m[1]
+  let l:repo = m[2]
```
Comment (skanehira) [inline autoload/gh/issues.vim]: We doesn't use prefix `l:` in function local; Cause when defining variables doesn't use any prefix in function local,  then will be implicity be function local scope as use prefix `l:`

File: autoload/gh/issues.vim [hunk 3]
```diff
@@ -194,17 +194,31 @@ function! gh#issues#new() abort
...
   let s:gh_issue_new = {
-        \ 'owner': m[1],
-        \ 'name': m[2],
-        \ 'branch': m[3],
+        \ 'owner': l:owner,
+        \ 'name': l:repo,
+        \ 'branch': '',
```
Comment (skanehira) [inline autoload/gh/issues.vim]: I think it's better to define default value `main`.

File: autoload/gh/issues.vim [hunk 4]
```diff
@@ -194,17 +194,31 @@ function! gh#issues#new() abort
...
+        \.then({-> gh#github#repos#files(s:gh_issue_new.owner, s:gh_issue_new.name, s:gh_issue_new.branch)})
         \.then(function('s:get_template_files'))
         \.then(function('s:open_template_list'))
         \.catch(function('s:get_template_error'))
+
+endfunction
+
+function! s:set_default_branch(resp)
+  if !has_key(a:resp.body, 'default_branch')
+    call gh#gh#error_message('not found default branch')
```
Comment (skanehira) [inline autoload/gh/issues.vim]: not found default branch, using "main" instead.

**Review by skanehira — CHANGES_REQUESTED** [review]

**Comment by skanehira** [comment]
@yoshio15 Thanks :)
I left some comments.

File: autoload/gh/github/repos.vim (hunk 1 above)
Comment (yoshio15) [reply autoload/gh/github/repos.vim]: thanks, fixed by [this](https://github.com/skanehira/gh.vim/pull/87/commits/572b042025678c5e9f15864dcce366db71a4dedf) commit.

File: autoload/gh/issues.vim (hunk 3 above)
Comment (yoshio15) [reply autoload/gh/issues.vim]: thank, i didn't know about that ^^
fixed by [this](https://github.com/skanehira/gh.vim/pull/87/commits/2e455667b44c10fa3871b4b4453ed4b4aa9cf658).

File: autoload/gh/issues.vim (hunk 2 above)
Comment (yoshio15) [reply autoload/gh/issues.vim]: thanks, good to know ^^
fixed by [this](https://github.com/skanehira/gh.vim/pull/87/commits/2e455667b44c10fa3871b4b4453ed4b4aa9cf658).

File: autoload/gh/issues.vim (hunk 3 above)
Comment (yoshio15) [reply autoload/gh/issues.vim]: fixed by [this](https://github.com/skanehira/gh.vim/pull/87/commits/e733b04dc85272156a6092880693747233679a7c) commit.

File: autoload/gh/issues.vim (hunk 4 above)
Comment (yoshio15) [reply autoload/gh/issues.vim]: fixed by [this](https://github.com/skanehira/gh.vim/pull/87/commits/e733b04dc85272156a6092880693747233679a7c) commit.

**Comment by yoshio15** [comment]
@skanehira 
thank you for reviewing! i updated this PR.

**Review by skanehira — APPROVED** [review]
LGTM
Thanks!

**Comment by skanehira** [comment]
@yoshio15 Sorry, Could you update help?
I forget it.

**Comment by yoshio15** [comment]
@skanehira 
oh, sorry... I forgot to update help document.
created new PR (https://github.com/skanehira/gh.vim/pull/91)

**Pull request merged**